grid/move_constructor test fails in 64-bit mode

Error message like this:
```
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc: In function ‘void test_periodic_cube()’:
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:90:9: error: expected nested-name-specifier before ‘periodic_face_pair’
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:90:9: error: ‘periodic_face_pair’ has not been declared
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:90:28: error: expected ‘;’ before ‘=’ token
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:90:28: error: expected primary-expression before ‘=’ token
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:91:76: error: expected primary-expression before ‘;’ token
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:92:15: error: ‘periodic_face_pair’ was not declared in this scope
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:92:33: error: template argument 1 is invalid
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:92:33: error: template argument 2 is invalid
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:92:53: error: invalid type in declaration before ‘;’ token
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc: In function ‘void test_periodic_cube() [with int dim = 2]’:
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:118:25:   instantiated from here
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:93:3: error: no matching function for call to ‘collect_periodic_faces(dealii::Triangulation<2, 2>&, int, int, int, int&)’
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:93:3: note: candidates are:
6108: /u/bangerth/p/deal.II/2/dealii/include/deal.II/grid/grid_tools.h:1464:3: note: template<class MeshType> void dealii::GridTools::collect_periodic_faces(const MeshType&, dealii::types::boundary_id, dealii::types::boundary_id, int, std::vector<dealii::GridTools::PeriodicFacePair<typename MeshType::cell_iterator> >&, const dealii::Tensor<1, MeshType:: space_dimension>&, const dealii::FullMatrix<double>&)
6108: /u/bangerth/p/deal.II/2/dealii/include/deal.II/grid/grid_tools.h:1498:3: note: template<class MeshType> void dealii::GridTools::collect_periodic_faces(const MeshType&, dealii::types::boundary_id, int, std::vector<dealii::GridTools::PeriodicFacePair<typename MeshType::cell_iterator> >&, const dealii::Tensor<1, MeshType:: space_dimension>&, const dealii::FullMatrix<double>&)
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:94:3: error: no matching function for call to ‘dealii::Triangulation<2, 2>::add_periodicity(int&)’
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:94:3: note: candidate is:
6108: /u/bangerth/p/deal.II/2/dealii/include/deal.II/grid/tria.h:3009:3: note: void dealii::Triangulation<dim, spacedim>::add_periodicity(const std::vector<dealii::GridTools::PeriodicFacePair<dealii::TriaIterator<dealii::CellAccessor<dim, spacedim> > > >&) [with int dim = 2, int spacedim = 2]
6108: /u/bangerth/p/deal.II/2/dealii/include/deal.II/grid/tria.h:3009:3: note:   no known conversion for argument 1 from ‘int’ to ‘const std::vector<dealii::GridTools::PeriodicFacePair<dealii::TriaIterator<dealii::CellAccessor<2, 2> > >, std::allocator<dealii::GridTools::PeriodicFacePair<dealii::TriaIterator<dealii::CellAccessor<2, 2> > > > >&’
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc: In function ‘void test_periodic_cube() [with int dim = 3]’:
6108: /u/bangerth/p/deal.II/2/dealii/tests/grid/move_constructor.cc:119:25:   instantiated from here
```

One of @danshapero 's recent patches probably caused this, though I don't want to imply he needs to fix it.

--- a/tests/grid/move_constructor.cc
+++ b/tests/grid/move_constructor.cc
@@ -1,6 +1,6 @@
 // ---------------------------------------------------------------------
 //
-// Copyright (C) 2003 - 2015 by the deal.II authors
+// Copyright (C) 2003 - 2016 by the deal.II authors
 //
 // This file is part of the deal.II library.
 //
@@ -87,8 +87,7 @@ void test_periodic_cube()
   Triangulation<dim> tria;
   GridGenerator::hyper_cube(tria, -1.0, 1.0, true);
 
-  using periodic_face_pair =
-    GridTools::PeriodicFacePair<typename Triangulation<dim>::cell_iterator>;
+  typedef GridTools::PeriodicFacePair<typename Triangulation<dim>::cell_iterator> periodic_face_pair;
   std::vector<periodic_face_pair> periodicity_vector;
   GridTools::collect_periodic_faces(tria, 0, 1, 0, periodicity_vector);
   tria.add_periodicity(periodicity_vector);

